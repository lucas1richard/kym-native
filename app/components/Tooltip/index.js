import styled from 'styled-components';

const Tooltip = styled.div`
  position: absolute;
  background: white;
  padding: 1rem;
  border-radius: 3px;
  border: 1px solid ${(props) => props.theme.mediumGray};
  bottom: ${(props) => !props.fromBelow && 'calc(100% + 15px)'};
  top: ${(props) => props.fromBelow && 'calc(100% + 15px)'};
  right: -1rem;
  box-shadow: 4px 14px 30px 0 rgba(0,0,0,0.1);
  min-width: ${(props) => props.minWidth};
  font-size: 0.75rem;
  text-align: left;
  ${''/* transform: translate3d(0, 0, 0); */}
  ${''/* transition: opacity 0.5s; */}
  z-index: 150;
  &:before, &:after {
    content: ' ';
    position: absolute;
    border-left: solid transparent;
    border-right: solid transparent;
  }
  &:after {
    content: ' ';
    border-top: ${(props) => !props.fromBelow && 'solid white'};
    border-bottom: ${(props) => props.fromBelow && 'solid white'};
    border-width: 7px;
    bottom: ${(props) => !props.fromBelow && '-7px'};
    top: ${(props) => props.fromBelow && '-7px'};
    right: ${(props) => props.right};
  }
  &:before {
    content: ' ';
    border-top: ${(props) => !props.fromBelow && `solid ${props.theme.mediumGray}`};
    border-bottom: ${(props) => props.fromBelow && `solid ${props.theme.mediumGray}`};
    border-width: 8px;
    bottom: ${(props) => !props.fromBelow && '-8px'};
    top: ${(props) => props.fromBelow && '-8px'};
    right: calc(${(props) => props.right} - 1px);
  }
`;

export default Tooltip;
