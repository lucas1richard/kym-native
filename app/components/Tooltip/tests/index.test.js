import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tooltip from '../';

describe('<Tooltip />', () => {
  const renderedComponent = (props = {}) => shallow(<Tooltip {...props} />);

  const props = {
    minWidth: '200px',
    right: '0px'
  };
  it('should render without blowing up', () => {
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
    props.fromBelow = true;
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });
});
