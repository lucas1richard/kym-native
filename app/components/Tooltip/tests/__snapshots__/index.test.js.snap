// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render without blowing up 1`] = `
<div
  className="sc-bdVaJa jqwEfz"
/>
`;

exports[`<Tooltip /> should render without blowing up 2`] = `
<div
  className="sc-bdVaJa cPhBqA"
/>
`;
